Chatkit/Chatkit/Chatkit.swift

```diff
@@ -1,6 +1,17 @@
 import Foundation
 import PusherPlatform
 
+public enum ChatkitError: String, LocalizedError {
```

Comment:
Happy to oblige on merging the two enums.  

Are you able to elaborate on your concerns about the enum being a `String`?  I'm wondering if this might actually _help_ customers because if they print the error they'll get a useful message rather than just enum case value (e.g. "connecting").

Reply:
Actually I am mistaken, serves me right for assuming I've just tested this in a playground and it still prints the enum value even if its a `String`.

Reply:
The enum needs to conform to `CustomStringConvertible` if its going to print anything but its case value.  Any issue with this...

```
/// An enumeration representing possible errors related to the Chatkit service.
public enum ChatkitError: Error {
    
    /// The error case representing an issue actioning the request because Chatkit is disconnected.
    case disconnected
    
    /// The error case representing an issue actioning the request because Chatkit is in the process of connecting.
    case connecting
    
    /// The error case representing an issue with the format of the provided instance locator.
    case invalidInstanceLocator
    
    /// The error case representing an issue with the format of the event received from Chatkit web service.
    case invalidEvent

}

extension ChatkitError: CustomStringConvertible {
    
    public var description: String {
        
        switch self {
        case .disconnected:
            return "The request cannot be actioned. Chatkit is disconnected."
        case .connecting:
            return "The request cannot be actioned. Chatkit is in the process of connecting."
        case .invalidInstanceLocator:
            return "An issue was encountered with the format of the provided instance locator."
        case .invalidEvent:
            return "An issue was encountered with the format of the event received from Chatkit web service."
        }
    }
    
}
```

Reply:
I've pushed this change.  I can always adjust if necessary.